pkg/controller/v1alpha1/localmodel/controller.go

```diff
@@ -18,6 +18,7 @@ limitations under the License.
 // +kubebuilder:rbac:groups=serving.kserve.io,resources=localmodelnodegroups,verbs=get;list;watch
 // +kubebuilder:rbac:groups=serving.kserve.io,resources=localmodelcaches,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=serving.kserve.io,resources=localmodelcaches/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=serving.kserve.io,resources=localmodelcaches/finalizers,verbs=update
```

Comment:
why finalizers are added in this PR?

upstream does not need the privilege, and I don't see any new code using finalizers.

Reply:
Without this rbac, the controllers would have the following error : 
```
{"level":"error","ts":"2025-08-30T19:42:16Z","logger":"v1alpha1Controllers.LocalModelNode","msg":"Failed to create
  job.","name":"","error":"jobs.batch \"xxxx\" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set
  finalizers on:
  ```